target/linux/ath79/files/include/mfd/rb4xx-cpld.h

```diff
@@ -0,0 +1,25 @@
+// SPDX-License-Identifier: GPL-2.0-only
+/*
+ * CPLD driver for the MikroTik RouterBoard 4xx series
+ *
+ * Copyright (C) 2008-2011 Gabor Juhos <juhosg@openwrt.org>
+ * Copyright (C) 2008 Imre Kaloz <kaloz@openwrt.org>
+ * Copyright (C) 2015 Bert Vermeulen <bert@biot.com>
+ * Copyright (C) 2020 Christopher Hill <ch6574@gmail.com>
+ *
+ * This file was based on the driver for Linux 2.6.22 published by
+ * MikroTik for their RouterBoard 4xx series devices.
+ */
+#include <linux/spi/spi.h>
+
+struct rb4xx_cpld {
+	struct spi_device *spi;
+
+	int (*rb4xx_cpld_write_nand)(struct rb4xx_cpld *self,
+				     const void *tx_buf, unsigned int len);
+	int (*rb4xx_cpld_read_nand)(struct rb4xx_cpld *self, void *rx_buf,
+				    unsigned int len);
+
+	int (*rb4xx_cpld_gpio_set_0_7)(struct rb4xx_cpld *self, u8 values);
+	int (*rb4xx_cpld_gpio_set_8)(struct rb4xx_cpld *self, u8 value);
```

Comment:
Purely cosmetic but I think you could make the code a bit more pleasant to read by skipping the `rb4xx_cpld_` prefix on these function pointers. Then in the code they can be used as e.g. `cpld->write_nand`. The `rb4xx_cpld` information is already included in the struct name.